File: web_service_ITDEL/app/Models/Pemesanan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Pemesanan extends Model
{
    use HasFactory;
    protected $table = 'pemesanan';
    protected $fillable = [
        'user_id', 'kaos_id', 'ukuran', 'jumlah_pesanan', 'total_harga','status_pembayaran'
    ];

    public function kaos()
    {
        return $this->belongsTo(Kaos::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: it_del/lib/Models/Pemesanan.dart
class Pemesanan {
  int id;
  int user_id;
  int kaos_id;
  int jumlah_pesanan;
  int total_harga;

  Pemesanan(
      {required this.id,
      required this.user_id,
      required this.kaos_id,
      required this.jumlah_pesanan,
      required this.total_harga,
      required});

  factory Pemesanan.fromJson(Map<String, dynamic> json) {
    return Pemesanan(
      id: json['id'],
      user_id: json['user_id'] ?? 0, // handle null if needed
      kaos_id: json['kaos_id'] ?? '',
      jumlah_pesanan: json['jumlah_pesanan'] ?? '',
      total_harga: json['total_harga'] ?? '',
    );
  }
}
